### PR TITLE
[MISC] Suppress oversee users in standby clusters

### PR DESCRIPTION
**Pull request by dragomirp**
Stop deleting managed users without a relation. This is necessary to keep all credentials working when there is a switchover to standby cluster.

### CODE REVIEW

File: src/relations/async_replication.py [hunk 1]
```diff
@@ -481,7 +482,7 @@ def is_primary_cluster(self) -> bool:
         return self.charm.app == self._get_primary_cluster()
 
     def _on_async_relation_broken(self, _) -> None:
-        if "departing" in self.charm._peers.data[self.charm.unit]:
+        if not self.charm._peers or "departing" in self.charm._peers.data[self.charm.unit]:
```
Comment (dragomirp) [inline src/relations/async_replication.py:485]: Seen it fail in CI.

File: tests/integration/ha_tests/test_async_replication.py [hunk 2]
```diff
@@ -218,6 +228,19 @@ async def test_async_replication(
...
+async def test_get_data_integrator_credentials(
+    ops_test: OpsTest,
+):
+    unit = ops_test.model.applications[DATA_INTEGRATOR_APP_NAME].units[0]
+    action = await unit.run_action(action_name="get-credentials")
+    result = await action.wait()
+    global data_integrator_credentials
+    data_integrator_credentials = result.results
+
+
```
Comment (dragomirp) [inline tests/integration/ha_tests/test_async_replication.py:243]: Just gets the credentials from data integrator.

File: src/relations/async_replication.py [hunk 3]
```diff
@@ -193,6 +193,7 @@ def _configure_standby_cluster(self, event: RelationChangedEvent) -> bool:
             filename = f"{POSTGRESQL_DATA_PATH}-{str(datetime.now()).replace(' ', '-').replace(':', '-')}.tar.gz"
             subprocess.check_call(f"tar -zcf {filename} {POSTGRESQL_DATA_PATH}".split())
             logger.warning("Please review the backup file %s and handle its removal", filename)
+        self.charm.app_peer_data["suppress-oversee-users"] = "true"
```
Comment (dragomirp) [inline src/relations/async_replication.py:196]: Set a flag when a cluster is made a standby

File: src/relations/postgresql_provider.py [hunk 4]
```diff
@@ -137,6 +137,8 @@ def oversee_users(self) -> None:
         if not self.charm.unit.is_leader():
             return
 
+        delete_user = "suppress-oversee-users" not in self.charm.app_peer_data
+
```
Comment (dragomirp) [inline src/relations/postgresql_provider.py:141]: If the cluster is or was a standby, it will stop deleting relation users.

**Comment by dragomirp** [comment]
Currently there's no way to clear up the flag. Action to clear it up was moved to https://github.com/canonical/postgresql-operator/pull/509

**Review by marceloneppel — APPROVED** [review]
It looks great! Thank you so much @dragomirp!

**Review by taurus-forever — APPROVED** [review]

**Pull request merged**